src/main/java/com/sngular/kloadgen/extractor/impl/SchemaExtractorImpl.java

```diff
@@ -68,11 +70,10 @@ public final Pair<String, List<FieldValueMapping>> flatPropertiesList(final Stri
     final List<FieldValueMapping> attributeList = new ArrayList<>();
     final HashMap<String, TypeElement> nestedTypes = new HashMap<>();
     String registryName = properties.getProperty(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME);
+    ParsedSchemaAdapter parsedSchemaAdapter = schemaParsed.getParsedSchemaAdapter();
+    schemaType = parsedSchemaAdapter.getType();
     if (Objects.nonNull(registryName) && registryName.equalsIgnoreCase(SchemaRegistryEnum.APICURIO.toString())) {
```

Comment:
Conditional can be simplified as SchemaRegistryEnum.APICURIO.toString().equalsIgnoreCase(registryName)